lib/helpers/emitter.js

```diff
@@ -51,4 +57,35 @@ exports.Emitter = class Emitter extends EventEmitter {
       .on('snapshot', onSnapshot)
       .on('end', onEnd);
   }
+
+  /**
+   * Filters the emitted events before passing to EventEmitter
+   * @param {string} eventName - event type emitted, must be one of VALID_EVENTS
+   */
+  async emit(eventName, ...args) {
+    if (!this.VALID_EVENTS.has(eventName)) throw new EmitterError(`The event type ${eventName} is not recognized`);
```

Comment:
What happens on the platform when you emit multiple pieces of data or multiple errors? Does the CLI have the same behaviour?

Reply:
Multiple pieces of data is in the test `manyEmits.js`
Re: multiple errors: the platform will stop after the first error. Added this functionality